diesel/src/pg/types/json.rs

```diff
@@ -0,0 +1,121 @@
+//! Support for JSON and `jsonb` values under PostgreSQL.
+
+extern crate serde_json;
+
+use std::io::prelude::*;
+use std::error::Error;
+
+use pg::Pg;
+use types::{self, ToSql, IsNull, FromSql};
+
+// The OIDs used to identify `json` and `jsonb` are not documented anywhere
+// obvious, but they are discussed on various PostgreSQL mailing lists,
+// including:
+//
+// https://www.postgresql.org/message-id/CA+mi_8Yv2SVOdhAtx-4CbpzoDtaJGkf8QvnushdF8bMgySAbYg@mail.gmail.com
+// https://www.postgresql.org/message-id/CA+mi_8bd_g-MDPMwa88w0HXfjysaLFcrCza90+KL9zpRGbxKWg@mail.gmail.com
+primitive_impls!(Json -> (serde_json::Value, pg: (114, 199)));
+primitive_impls!(Jsonb -> (serde_json::Value, pg: (3802, 3807)));
```

Comment:
Seconded. This seems like something that should be added in the pq-sys bindings, too.